src/components/WebCLI/WebCLI.tsx

```diff
@@ -127,22 +127,53 @@ const WebCLI = ({
     const conn = new Connection({
```

Comment:
We shouldn't create a new connection if there is an active one. This shouldn't happen, but it's probably worth adding the check just in case.

```suggestion
    // If we have an active WebSocket connection then don't create a new one.
    if (connection.current?.isActive()) {
      return;
    }
    const conn = new Connection({
```